c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -2177,19 +2177,20 @@ void MfxC2DecoderComponent::DoWork(std::unique_ptr<C2Work>&& work)
             if (it != m_pendingWorks.end()) {
                 work = std::move(it->second);
                 m_pendingWorks.erase(it);
+
+                if (work) {
```

Comment:
Could you please explain why we move the logic here?

Thanks,
Tianmi

Reply:
I have updated it.  Please review again.

Thanks,
Hao